### PR TITLE
Integrate User Service with Dashboard

### LINKED ISSUE #45: Integrate User Service with frontend dashboard
## What to do?
- update the api wrapper for user service
- update any hardcoded functions and variables
- update the dashboard page/components to consume the API
- fix bugs (if any)

### PR DESCRIPTION
**Pull request by tlyi**
# Pull Request

## Description

- Update User API Wrappers (done by @tryyang2001)
- Remove hardcoded values from frontend components
- Implemented an Auth Provider so that components can easily access current User

## Related Issue(s)

Closes #45 

## Changes Made

- Currently, AuthProvider uses a fixed id, but all other user profile details are retrieved from backend (I will integrate this with @weidak 's Login component + Auth provider (once it is completed))
- This is how to use the AuthContext. You can place this block of code on top of any components that require user details. The context will re-retrieve user details from backend every time the page reloads.

```typescript
 //modify specific attributes according to what the component requires
const { user: { image, name }, } = useAuthContext();
``` 

## Checklist
- [x] I have checked that the changes included in the PR are intended to merge to `master` or any destination branch.
- [x] I have verified that the new changes do not break any existing functionalities, unless the new changes are intended and have approved by the team.
- [x] I will take care of the merging and delete the side-branch after the PR is merged.

## Additional Notes/References

@c0j0s I have updated the matching card to retrieve the user preferences from the context as that current wrapper is outdated, you might wanna merge this first, sorry for the inconvenience!!

### CODE REVIEW

**Review by c0j0s — APPROVED** [review]
Sure no problem, LGTM

File: backend/user-service/src/controllers/handlers/get-handler.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const getUserById = async (request: Request, response: Response) => {
         id: userId,
       },
       include: {
-        preferences: {
+        Preferences: {
```
Comment (tryyang2001) [inline backend/user-service/src/controllers/handlers/get-handler.ts:32]: Oh weird, the `preferences` should not be capitalised here. Refer to `schema.prisma` in the user service, it should be all lowercase.

**Comment by tryyang2001** [comment]
I will suggest we pause merging this PR first before we merge this [PR](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g05/pull/51), since I believe @weidak has started working on refactoring the `user_api_wrapper`, so he should have the latest version.

File: backend/user-service/src/controllers/handlers/get-handler.ts (hunk 1 above)
Comment (tlyi) [reply backend/user-service/src/controllers/handlers/get-handler.ts:32]: Hmm I'm not sure, when I tried to run the code there was an error and this was the suggested fix so I just changed it 😅 Did you manage to run it with lowercase?

Comment (tlyi) [reply backend/user-service/src/controllers/handlers/get-handler.ts:32]: Ahh I see I see, okay thank you! Will resolve it when merging to the branch :)

**Pull request merged**